source/neuropod/multiprocess/mq/ipc_message_queue_impl.hh

```diff
@@ -163,6 +198,7 @@ IPCMessageQueue<UserPayloadType>::IPCMessageQueue(const std::string &control_que
       recv_queue_(detail::make_recv_queue<UserPayloadType>(control_queue_name_, type)),
       heartbeat_controller_(stdx::make_unique<detail::HeartbeatController>(*this)),
       transferrable_controller_(stdx::make_unique<detail::TransferrableController>()),
+      lost_heartbeat_(false),
```

Comment:
I don't see a full picture, maybe you checked this already.

This will be set to true on failure and never reset again. I guess the idea is that the given Neuropod object is not re-usable (that is fine). 

What happens with next "infer", isn't it causing other crash? Next infer may happen because of race, when it happend before Parent received exception and can stop serving.

Reply:
> I guess the idea is that the given Neuropod object is not re-usable (that is fine).

Yes, this was the case before and continues to be the case. Previously, a disconnect would cause an error to be thrown in another thread (and take down the whole application). Now, a disconnect will throw an error in the current thread, but the model is no longer usable. 

> Next infer may happen because of race, when it happend before Parent received exception and can stop serving.

As we've talked about a few times before, `infer` is _not_ threadsafe. `infer` should only be called by one thread at a time. If you're using a single model across threads with locking, I'd recommend unloading the model when you get a disconnect exception.

If you do call `infer` again, it will throw an error again (after potentially blocking for a bit).